extras/kinesis-asl/src/test/scala/org/apache/spark/streaming/kinesis/KinesisStreamSuite.scala

```diff
@@ -268,9 +270,9 @@ abstract class KinesisStreamTests(aggregateTestData: Boolean) extends KinesisFun
 
     // Verify that the recomputed RDDs are KinesisBackedBlockRDDs with the same sequence ranges
     // and return the same data
-    val times = collectedData.keySet
+    val times = collectedData.synchronized { collectedData.keySet }
```

Comment:
I think you'd either have to make a copy of the key set being returned, or synchronize the entire `foreach` block below. The set is (I believe) backed by the collection and can be modified during iteratio.